flowey/flowey_core/src/pipeline/artifact.rs

```diff
@@ -58,7 +59,12 @@ use std::path::Path;
 /// - subdir/
 ///   - file2.exe
 /// ```
-pub trait Artifact: Serialize + DeserializeOwned {}
+pub trait Artifact: Serialize + DeserializeOwned {
+    /// If present, the artifact should consist of a tar.gz file containing the
+    /// contents of the artifact. This is used when the artifact contents are
+    /// too large for Azure DevOps to handle directly.
```

Comment:
Careful about writing docs that elevate one backend over any other. This feature is - presumably - functional across all backends (gh actions, and local included). If so - having a clearer delineation in the docs between "here is what this does", and "here is why you may want to use, depending on what backend you're targeting" is good